### PR TITLE
Vec 231 0.9.0

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
* Added a --credentials flags
* Changed name of --index-metadata to --index-labels
* Add index update command
* Added new params to index create command
* Added new params to index ls command
* Updated tests to 0.9.0

avs-client-go will need a PR merged before tests will pass

### CODE REVIEW

File: cmd/indexGC.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		PreRunE: func(_ *cobra.Command, _ []string) error {
+			return checkSeedsAndHost()
+		},
+		RunE: func(_ *cobra.Command, _ []string) error {
+			debugFlags := indexGCFlags.clientFlags.NewSLogAttr()
+			logger.Debug("parsed flags",
+				append(debugFlags,
+					slog.String(flags.Namespace, indexGCFlags.namespace),
+					slog.String(flags.IndexName, indexGCFlags.indexName),
+					slog.Time(flags.HnswMaxMemQueueSize, indexGCFlags.cutoffTime.Time()),
```
Comment (dwelch-spike) [inline cmd/indexGC.go:64]:
```suggestion
					slog.Time(flags.CutoffTime, indexGCFlags.cutoffTime.Time()),
```

File: cmd/indexUpdate.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	hnswHealer:          *flags.NewHnswHealerFlags(),
+	hnswMerge:           *flags.NewHnswMergeFlags(),
+}
+
+func newIndexUpdateFlagSet() *pflag.FlagSet {
+	flagSet := &pflag.FlagSet{}
+	flagSet.BoolVarP(&indexUpdateFlags.yes, flags.Yes, "y", false, commonFlags.DefaultWrapHelpString("When true do not prompt for confirmation."))                                           //nolint:lll // For readability
+	flagSet.StringVarP(&indexUpdateFlags.namespace, flags.Namespace, "n", "", commonFlags.DefaultWrapHelpString("The namespace for the index."))                                             //nolint:lll // For readability
+	flagSet.StringVarP(&indexUpdateFlags.indexName, flags.IndexName, "i", "", commonFlags.DefaultWrapHelpString("The name of the index."))                                                   //nolint:lll // For readability
+	flagSet.StringToStringVar(&indexUpdateFlags.indexMeta, flags.IndexLabels, nil, commonFlags.DefaultWrapHelpString("The distance metric for the index."))                                  //nolint:lll // For readability
```
Comment (dwelch-spike) [inline cmd/indexUpdate.go:41]: Double checking that this is the distance metric for the index, the names of the flags "Meta" and "Labels" don't make this clear to me.

File: cmd/indexUpdate.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+			debugFlags = append(debugFlags, indexUpdateFlags.hnswBatch.NewSLogAttr()...)
+			debugFlags = append(debugFlags, indexUpdateFlags.hnswCache.NewSLogAttr()...)
+			debugFlags = append(debugFlags, indexUpdateFlags.hnswHealer.NewSLogAttr()...)
+			debugFlags = append(debugFlags, indexUpdateFlags.hnswMerge.NewSLogAttr()...)
+			logger.Debug("parsed flags",
+				append(debugFlags,
+					slog.Bool(flags.Yes, indexUpdateFlags.yes),
+					slog.String(flags.Namespace, indexUpdateFlags.namespace),
+					slog.String(flags.IndexName, indexUpdateFlags.indexName),
+					slog.String(flags.HnswMaxMemQueueSize, indexUpdateFlags.hnswMaxMemQueueSize.String()),
```
Comment (dwelch-spike) [inline cmd/indexUpdate.go:87]: No log for flags.IndexLabels ?

File: test.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    do
+    if [ "$line" != "${TAG}" ]
+    then
+        if [[ $line =~ ^${TAG}- ]]
+        then
+        echo "Removing $line"
+        gh release delete $line -R github.com/aerospike/asvec --yes --cleanup-tag
+        fi
+    fi
+    done
```
Comment (dwelch-spike) [inline test.sh:14]: This looks release related, why is it called test.sh?

**Review by dwelch-spike — COMMENTED** [review]
A few comments and nitpicks. Otherwise LGTM

File: cmd/flags/optionals.go [hunk 5]
```diff
@@ -49,6 +53,53 @@ func (f *Uint32OptionalFlag) String() string {
...
+	return err
+}
+
+func (f *Float32OptionalFlag) Type() string {
+	return "float32"
+}
+
+func (f *Float32OptionalFlag) String() string {
+	if f.Val != nil {
+		return strconv.FormatUint(uint64(*f.Val), 10)
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply cmd/flags/optionals.go:97]: good catch

File: cmd/indexUpdate.go (hunk 2 above)
Comment (jdogmcsteezy) [reply cmd/indexUpdate.go:41]: Yes, I missed this rename

File: cmd/indexUpdate.go (hunk 3 above)
Comment (jdogmcsteezy) [reply cmd/indexUpdate.go:87]: added

File: test.sh (hunk 4 above)
Comment (jdogmcsteezy) [reply test.sh:14]: removed

**Review by dwelch-spike — APPROVED** [review]
LGTM

**Pull request merged**